### ISSUE TITLE
crash: using oxlint --type-aware goroutine crash

### ISSUE THREAD

**Issue by rubnogueira** [oxc-project/tsgolint#574]
I opened #297 in the past, decided to try it, and it worked, but now when I was integrating it into our codebase, I got the crash:

### How to trigger
❯ oxlint --type-aware

oxlint without --type-aware flag works great.

---
❯ oxlint --version
Version: 1.38.0

❯ tsgo --version
Version 7.0.0-dev.20260109.1

Apparently tsgolint doesn't have the version flag, a suggestion here as well 😄 
tsgolint - 0.11.0

---

❯ which oxlint
/Users/myuser/.local/share/mise/installs/npm-oxlint/1.38.0/bin/oxlint

❯ which tsgolint
/Users/myuser/.local/share/mise/installs/npm-oxlint-tsgolint/0.11.0/bin/tsgolint

❯ which tsgolint
/Users/myuser/.local/share/mise/installs/npm-typescript-native-preview/7.0.0-dev.20260109.1/bin/tsgo

### Trace:

```
2026/01/10 12:51:34.894074 Unmatched files found: ...<hundreds of files>..., /Users/myuser/monorepo/libs/myproject/src/file1.ts, /Users/myuser/monorepo/libs/myproject/src/file2.ts, /Users/myuser/monorepo/libs/myproject/src/file3.ts, /Users/myuser/monorepo/libs/myproject/src/file4.ts, /Users/myuser/monorepo/libs/myproject/src/file5.ts' to be in program '/Users/myuser/monorepo/libs/myproject/tsconfig.json'

goroutine 1 [running]:
github.com/typescript-eslint/tsgolint/internal/linter.RunLinter(0x0, {0x14000024004, 0x4f}, {0x140001fe6c0?, {0x14000e0e280?, 0x14000133ac0?, 0x10333dff8?}}, 0x8, {0x10334a700, 0x140001248e0}, ...)
        /home/runner/work/tsgolint/tsgolint/internal/linter/linter.go:111 +0xcec
main.runHeadless({0x14000132060, 0x0, 0x0})
        /home/runner/work/tsgolint/tsgolint/cmd/tsgolint/headless.go:353 +0xa3c
main.runMain()
        /home/runner/work/tsgolint/tsgolint/cmd/tsgolint/main.go:355 +0x84
main.main()
        /home/runner/work/tsgolint/tsgolint/cmd/tsgolint/main.go:567 +0x1c
```

I have the `OXLINT_TSGOLINT_PATH` env variable set from the early tests and I'm not sure if it's relevant anymore.

**Comment by camchenry** [comment]
Closed via https://github.com/oxc-project/tsgolint/pull/577

**Comment by no-yan** [comment]
Thanks for the excellent reproduction!
Unfortunately, circular references aren't supported by TypeScript. The reason you didn't see an error is likely due to how it was run.

Project references are meant for build mode (tsc -b), and running it that way results in a circular graph error (TS6202).

I’ve created a PR for reproduction: http://github.com/rubnogueira/oxlint-typeaware-circular-repro/pull/1

Once #686 is merged, we should see a much clearer error message.
